main.tf

```
@@ -0,0 +1,377 @@
+  count = var.store_secret_in_key_vault ? 1 : 0
+
+  name         = "${var.name}-${var.admin_username}-${lower(var.authentication_type)}"
+  content_type = var.authentication_type
+  key_vault_id = var.key_vault_id
+  value        = coalesce(local.admin_password, local.admin_ssh_private_key)
+}
+
+resource "azurerm_user_assigned_identity" "this" {
+  count = local.create_identity ? 1 : 0
+
+  name                = "id-${trimprefix(var.name, "vm-")}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
+}
+
+resource "azurerm_managed_disk" "this" {
+  for_each = { for element in local.data_disks : element.name => element }
+
+  name                = each.value.name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
+
+  storage_account_type = each.value.storage_account_type
+  create_option        = each.value.create_option
+  disk_size_gb         = each.value.disk_size_gb
+}
+
+resource "azurerm_virtual_machine_data_disk_attachment" "this" {
+  for_each = { for element in local.data_disks : element.name => element }
+
+  managed_disk_id    = azurerm_managed_disk.this[each.key].id
+  virtual_machine_id = local.virtual_machine.id
+  lun                = each.value.lun
+  caching            = each.value.caching
+}
+
+resource "azurerm_virtual_machine_extension" "this" {
+
+  for_each = { for element in [
+    {
+      name                       = "NetworkWatcherAgent"
+      publisher                  = "Microsoft.Azure.NetworkWatcher"
+      type                       = local.is_windows ? "NetworkWatcherAgentWindows" : (local.is_linux ? "NetworkWatcherAgentLinux" : null)
+      type_handler_version       = "1.0"
+      auto_upgrade_minor_version = true
+      automatic_upgrade_enabled  = true
+    },
+    {
+      name                       = "AzureMonitorAgent"
+      publisher                  = "Microsoft.Azure.Monitor"
+      type                       = local.is_windows ? "AzureMonitorWindowsAgent" : (local.is_linux ? "AzureMonitorLinuxAgent" : null)
+      type_handler_version       = "1.0"
+      auto_upgrade_minor_version = true
+      automatic_upgrade_enabled  = true
+    },
+    {
+      name                       = "AzurePolicy"
+      publisher                  = "Microsoft.GuestConfiguration"
+      type                       = local.is_windows ? "ConfigurationforWindows" : (local.is_linux ? "ConfigurationForLinux" : null)
+      type_handler_version       = "1.0"
+      auto_upgrade_minor_version = true
+      automatic_upgrade_enabled  = true
+    },
+    local.is_windows ? {
+      name                       = "AntiMalware"
+      publisher                  = "Microsoft.Azure.Security"
+      type                       = "IaaSAntimalware"
+      type_handler_version       = "1.0"
+      auto_upgrade_minor_version = true
+      automatic_upgrade_enabled  = false
+    } : null
```

Kindly request the boot attestation extension here as default.

If Secure Boot and vTPM is enabled the Integrity monitoring should be enabled as well. ref: https://learn.microsoft.com/en-us/azure/virtual-machines/boot-integrity-monitoring-overview?tabs=portal

The extension was also availible via old module: https://github.com/cloudeteer/tf-mod-lib/blob/55c545f7539de5e92438dfdad0379cae8933e878/azure/windows_virtual_machine/windows_virtual_machine.tf#L501

```suggestion
    } : null,
    varsecure_boot_enabled && var.vtpm_enabled ? {
      name                       = "GuestAttestation"
      publisher                  = local.is_windows ? "Microsoft.Azure.Security.WindowsAttestation" : "Microsoft.Azure.Security.LinuxAttestation"
      type                       = "GuestAttestation"
      type_handler_version       = "1.0"
      auto_upgrade_minor_version = true
      automatic_upgrade_enabled  = true
    } : null,
```

---

I missed checking for the extensions in the previous Windows vm module, only considered the Linux vm module. Thanks for this!

---

Will be covered in issue #6.